concrete_datastore/api/v1/views.py

```diff
@@ -408,6 +464,79 @@ def post(self, request, *args, **kwargs):
         return Response(data=serializer.data, status=HTTP_200_OK)
 
 
+class SecureLoginCodeApiView(generics.GenericAPIView):
+    """
+    this view is used to login the user with Secure Login
+    with email and code
+    """
+
+    serializer_class = SecureLoginCodeSerializer
+    api_namespace = DEFAULT_API_NAMESPACE
+
+    def __init__(self, api_namespace, *args, **kwargs):
+        self.api_namespace = api_namespace
+        super().__init__(*args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        ip = get_client_ip(request)
+        now = pendulum.now('utc').format(settings.LOGGING['datefmt'])
+        user = self.request.user
+        base_message = f"[{now}|{ip}|{user}|AUTH] "
+
+        serializer = self.serializer_class(data=request.data)
+        if not serializer.is_valid():
+            return Response(
+                data={
+                    "message": serializer.errors,
+                    "_errors": ["INVALID_DATA"],
+                },
+                status=HTTP_400_BAD_REQUEST,
+            )
+        email = serializer.validated_data['email']
+        code = serializer.validated_data['code']
+        try:
+            secure_connect_code = SecureConnectCode.objects.get(
+                user__email=email, value=code
+            )
+        except ObjectDoesNotExist:
+            log_request = base_message + (
+                f"Secure login attempt failed: invalid code"
+            )
+            logger_api_auth.info(log_request)
+            return Response(
+                data={'message': 'Invalid code', "_errors": ["INVALID_CODE"]},
+                status=HTTP_401_UNAUTHORIZED,
+            )
+        user = secure_connect_code.user
```

Comment:
that line should be in a `else` statement of the try/except block for more readibility

Reply:
Added into an else statement